test/ng/directive/asSpec.js

```diff
@@ -0,0 +1,87 @@
+'use strict';
+
+describe('as', function() {
+
+  it('should bind in the current scope the controller of a component', function() {
+    var expected = {some: 'value'};
+
+    angular.module('my', []).component('myComponent', {
+      template: 'foo',
+      controller: function() {
+        this.property = expected;
+      }
+    });
+    module('my');
+
+    inject(function($compile, $rootScope) {
+      var scope = $rootScope.$new();
+      var $ctrl = {undamaged: true};
+      scope.$ctrl = $ctrl;
+      dealoc($compile('<my-component as="$ctrl.myComponent"></my-component>')(scope));
+
+      expect(scope.$ctrl).toBe($ctrl);
+      expect(scope.$ctrl.undamaged).toBe(true);
+      expect($ctrl.myComponent).not.toBeUndefined();
+      expect($ctrl.myComponent && $ctrl.myComponent.property).toBe(expected);
+      scope.$destroy();
```

Comment:
Ops! Right! Each test have a new shiny $rootScope, so not even new required.

About dealoc a I do not have clear if I must add or remove it, I found the following:

- https://github.com/angular/material/issues/2250
- https://github.com/angular/angular.js/blob/1061c56fe159691112bc81ca34573ea9681a7255/test/ng/compileSpec.js#L738-L748

And tests from:

- https://github.com/angular/angular.js/blob/1061c56fe159691112bc81ca34573ea9681a7255/test/ng/compileSpec.js#L10001-L10015
are using the global "element" variable cleaned by:
- https://github.com/angular/angular.js/blob/1061c56fe159691112bc81ca34573ea9681a7255/test/ng/compileSpec.js#L145

But I'm not sure if the main afterEach is executed after the whole describe, or after each piece. Or may be it is a bug.

But if you are confident that it is not necessary I'll remove it.